rfcs/023-images-endpoint/README.md

```diff
@@ -0,0 +1,48 @@
+# RFC 023: Images endpoint
+
+**Status:** :construction: Draft :building_construction:
+
+**Last updated:** 27/04/2020
+
+## Context and requirements
+
+- Images are now entities in the catalogue API
+- We want to be able to search for images, and to fetch specified images
+- The images endpoint should behave similarly to the works endpoint
+
+## Proposal
+
+The initial images endpoint will be in **beta**, and therefore will be liable to have breaking changes made without a version bump.
+
+### Image display model
+
+```
+{
+  "id": String,
+  "location": DigitalLocation,
+  "parentWork": String
```

Comment:
nit: language

Similar to the language [being discussed here](https://github.com/wellcomecollection/storage-service/pull/486#discussion_r416408216), should this just be a `workLink` or similar, and in this case the way of expressing this link is a string.

This implies that the work has to exist for the image to exist, which, while is practically true as these images are generated from works, is that what we're trying to imply?

Reply:
Good point, I think perhaps we actually _do_ want to imply this? I remember it being discussed before but can't remember much about that...

Reply:
Would we say that the images are children of the work?

Either way - could we type this as a `case class WorkLink(id: String)` as I think this will start to be useful in things like the concepts work?

---

Sideline:
As a point, a nice piece of language spotted through our ontologies [is `partOf`](https://github.com/wellcomecollection/platform/search?q=part+of&unscoped_q=part+of).
Not sure it actually applies here.

Reply:
I don't think this link should be expressed as a string, it should be an object that contain potentially include metadata from the work. The property name should also express the relationship, which I'm not sure should be parent, but still pondering that.

Reply:
I wonder whether something like `sourceWork` is better? It's not really a parent/child relationship. That or `relatedWork` but that feels not strong enough and a bit generic. Doesn't say how it's related.

Reply:
I like `sourceWork`

Reply:
And all images will come from a work and no other sources?

Reply:
Good point, how about something like
```scala
trait ImageSource {
  val id: Id
}

case class SourceWork(
  id: Id,
  sourceType: String = "Work"
) extends ImageSource

case class DisplayImage(
  ...,
  source: ImageSource
)
```

Reply:
Looks ✨ 

`SourceWork` will definitely come in handy elsewhere (if we abstract `ImageSource` to `Source` or something).